### ISSUE TITLE
`allow_zero_lintable_files: true` doesn't work when added to a parent config that's loaded by a child config

### ISSUE THREAD

**Issue by vakhidbetrakhmadov** [realm/SwiftLint#6240]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

Hi, `allow_zero_lintable_files: true` doesn't work when added to a parent config that's loaded by a child config.
It only works if `allow_zero_lintable_files: true` is specified in the child config.

```bash
$ swiftlint lint --config Foo/Bar/.swiftlint.yml --force-exclude --no-cache --quiet Foo/Bar/Tests/DeepLinkParserTests.swift
Error: No lintable files found at paths: 'Foo/Bar/Tests/DeepLinkParserTests.swift'
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)
`0.61.0`

* Xcode version (run `xcodebuild -version` to be sure)
```
Xcode 16.4
Build version 16F6
```

* Installation method used (Homebrew, CocoaPods, building from source, etc)
`Universal macOS Binary`

* Configuration file:

```
# Foo/Bar/.swiftlint.yml

parent_config: ../../.swiftlint_base.yml

excluded:
  - Tests
```

```
# .swiftlint_base.yml

allow_zero_lintable_files: true
```

**Comment by SimplyDanny** [comment]
This needs a good concept first. `allow_zero_lintable_files` is only one example of an option that currently ignores the parent. A fix should be done for all relevant options consistently though. Since this would be a breaking change, it needs to be documented on the changelog well.
